src/main/java/com/app/onlinebookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,117 @@
+package com.app.onlinebookstore.service.impl;
+
+import com.app.onlinebookstore.dto.CreateOrderRequestDto;
+import com.app.onlinebookstore.dto.OrderDto;
+import com.app.onlinebookstore.dto.OrderItemDto;
+import com.app.onlinebookstore.dto.OrderStatusDto;
+import com.app.onlinebookstore.exaption.EntityNotFoundException;
+import com.app.onlinebookstore.mapper.OrderItemMapper;
+import com.app.onlinebookstore.mapper.OrderMapper;
+import com.app.onlinebookstore.model.CartItem;
+import com.app.onlinebookstore.model.Order;
+import com.app.onlinebookstore.model.OrderItem;
+import com.app.onlinebookstore.model.ShoppingCart;
+import com.app.onlinebookstore.model.Status;
+import com.app.onlinebookstore.model.User;
+import com.app.onlinebookstore.repository.OrderItemRepository;
+import com.app.onlinebookstore.repository.OrderRepository;
+import com.app.onlinebookstore.repository.ShoppingCartRepository;
+import com.app.onlinebookstore.service.OrderService;
+import com.app.onlinebookstore.service.UserService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final UserService userService;
+
+    @Override
+    public OrderDto createOrder(CreateOrderRequestDto requestDto, Authentication authentication) {
+        User currentUser = (User) authentication.getPrincipal();
+        ShoppingCart shoppingCart = shoppingCartRepository

```

Comment:
For retrieving shopping cart and handling notFoundEx should be responsible shoppingCartService